lib/bolt/plugin/puppet_connect_data.rb

```diff
@@ -3,12 +3,32 @@
 module Bolt
   class Plugin
     class PuppetConnectData
+      INPUT_DATA_VAR = 'PUPPET_CONNECT_INPUT_DATA'
+
       def initialize(context:, **_opts)
-        puppet_connect_data_yaml_path = File.join(context.boltdir, 'puppet_connect_data.yaml')
+        data_path = nil
+        if ENV.key?(INPUT_DATA_VAR)
+          @input_data_path = ENV[INPUT_DATA_VAR]
+          data_path = @input_data_path
+        else
+          data_path = File.join(context.boltdir, 'puppet_connect_data.yaml')
+        end
+
         @data = Bolt::Util.read_optional_yaml_hash(
-          puppet_connect_data_yaml_path,
-          'puppet_connect_data.yaml'
+          data_path,
+          File.basename(data_path)
         )
+
+        if @input_data_path
+          # Validate that the data does not contain any plugin-reference
+          # values
+          @data.each do |k, v|
```

Comment:
This should check for nested values:

```yaml
mykey:
  another_key:
    _plugin: myplugin
```

I think there are some helpers for that in `Bolt::Util`.

Reply:
Thanks, updated.